core/domain/topic_services_test.py

```diff
@@ -628,24 +743,38 @@ def test_publish_and_unpublish_topic(self):
         topic_rights = topic_services.get_topic_rights(self.TOPIC_ID)
         self.assertFalse(topic_rights.topic_is_published)
         topic_services.publish_topic(self.TOPIC_ID, self.user_id_admin)
-
         with self.assertRaisesRegexp(
-            Exception,
-            'The user does not have enough rights to unpublish the topic.'):
-            topic_services.unpublish_topic(self.TOPIC_ID, self.user_id_a)
+            Exception, 'The topic is already published.'):
```

Comment:
Split all these cases into multiple functions with one function for each case. See the guidelines [here](https://github.com/oppia/oppia/wiki/Writing-backend-tests#guidelines-for-writing-good-tests) (Rule 2 to be specific)